components/camel-kafka/src/main/java/org/apache/camel/component/kafka/TaskHealthState.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.kafka;
+
+import java.util.Properties;
+
+import org.apache.camel.util.TimeUtils;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+
+/*
+ * This one is used to avoid exposing methods from this class to the health checker. Fields and methods used
+ * used to build an instance of this class, must be made thread-safe (i.e.: most importantly, read fields
+ * should be marked as volatile).
+ */
+public class TaskHealthState {
```

Comment:
private package instead of public as it seems to be an implementation detail?

Reply:
+1 from me, but let's leave it like this for now. I am considering moving the whole set of health check code into a separate package (or, at least, within each of `consumer` and `producer` packages).